lib/ParamedicSauceLabs.js

```diff
@@ -18,538 +18,553 @@
-        retryDelay: util.WD_RETRY_DELAY,
-        retries: util.WD_RETRIES
-    });
-
-    var driver = wd.promiseChainRemote(util.SAUCE_HOST, util.SAUCE_PORT, user, key);
-    return driver
-        .init(caps)
-        .then(function () {
-            clearInterval(spamDots);
-            process.stdout.write('\n');
-        }, function (error) {
-            clearInterval(spamDots);
-            process.stdout.write('\n');
-            throw (error);
+            if (!found) {
+                logger.warn('Can not find saucelabs job. Logs and video will be unavailable.');
+            }
+            d.resolve();
         });
-};
 
-ParamedicSauceLabs.prototype.connectSauceConnect = function () {
-    var self = this;
-    var isBrowser = self.config.getPlatformId() === util.BROWSER;
+        return d.promise;
+    }
+
+    getSauceCaps () {
+        this.runner.sauceBuildName = this.runner.sauceBuildName || this.config.getBuildName();
+        let caps = {
+            name: this.runner.sauceBuildName,
+            idleTimeout: '100', // in seconds
+            maxDuration: utilities.SAUCE_MAX_DURATION,
+            tunnelIdentifier: this.config.getSauceTunnelId()
+        };
+
+        switch (this.platformId) {
+        case utilities.ANDROID:
+            caps.platformName = 'Android';
+            caps.appPackage = 'io.cordova.hellocordova';
+            caps.appActivity = 'io.cordova.hellocordova.MainActivity';
+            caps.app = 'sauce-storage:' + this.getAppName();
+            caps.deviceType = 'phone';
+            caps.deviceOrientation = 'portrait';
+            caps.appiumVersion = this.config.getSauceAppiumVersion();
+            caps.deviceName = this.config.getSauceDeviceName();
+            caps.platformVersion = this.config.getSaucePlatformVersion();
+            break;
+
+        case utilities.IOS:
+            caps.platformName = 'iOS';
+            caps.autoAcceptAlerts = true;
+            caps.waitForAppScript = 'true;';
+            caps.app = 'sauce-storage:' + this.getAppName();
+            caps.deviceType = 'phone';
+            caps.deviceOrientation = 'portrait';
+            caps.appiumVersion = this.config.getSauceAppiumVersion();
+            caps.deviceName = this.config.getSauceDeviceName();
+            caps.platformVersion = this.config.getSaucePlatformVersion();
+            break;
+
+        case utilities.BROWSER:
+            caps.browserName = this.config.getSauceDeviceName() || 'chrome';
+            caps.version = this.config.getSaucePlatformVersion() || '45.0';
+            caps.platform = caps.browserName.indexOf('Edge') > 0 ? 'Windows 10' : 'macOS 10.13';
+            // setting from env.var here and not in the config
+            // because for any other platform we don't need to put the sauce connect up
+            // unless the tunnel id is explicitly passed (means that user wants it anyway)
+            if (!caps.tunnelIdentifier && process.env[utilities.SAUCE_TUNNEL_ID_ENV_VAR]) {
```

Comment:
Maybe I could be this this. Instead of checking `if (!caps.tunnelIdentifier)` twice.

```
if (!caps.tunnelIdentifier) {
  if (process.env[utilities.SAUCE_TUNNEL_ID_ENV_VAR]) {
    caps.tunnelIdentifier = process.env[utilities.SAUCE_TUNNEL_ID_ENV_VAR];
  } else {
    throw new Error('Testing browser platform on Sauce Labs requires Sauce Connect tunnel. Please specify tunnel identifier via --sauceTunnelId');
  }
}

Reply:
This may lower the cyclomatic complexity but will look into apply this in a later PR.